### PR TITLE
Add `atmos describe affected` CLI command. Update docs website

### PR DESCRIPTION
**Pull request by aknysh**
## what
* Add `atmos describe affected` CLI command
* Update docs website (https://pr-274.atmos-docs.ue2.dev.plat.cloudposse.org/cli/commands/describe/affected/)

## why
* The command `atmos describe affected` produces a list of the affected Atmos components and stacks given two Git commits. The command compares the  final component sections (after all imports and deep-merging) for all Atmos components in all stacks, and produces a list of affected (changed) components in the stacks. The command also checks the changed files b/w the two commits and checks if the Terraform/Helmfile component folders are changed.

For the first commit, the command assumes that the repo root is a Git checkout (and throws an error if the repo is not a Git repository, does not have `.git` folder).

The second commit is specified on the command line using the `--ref` and `--sha` flags.
The `--ref` flag supports all standard Git References (https://git-scm.com/book/en/v2/Git-Internals-Git-References).

the `ref` will be the default branch (e.g. `main`) and the commit SHA will point to the `HEAD` of the branch.

```shell
atmos describe affected  --verbose=true

Cloning repo 'https://github.com/cloudposse/atmos' into the temp dir '/var/folders/g5/lbvzy_ld2hx4mgrgyp19bvb00000gn/T/16710538942745756531'

Checking out the HEAD of the default branch

Enumerating objects: 4138, done.
Counting objects: 100% (1080/1080), done.
Compressing objects: 100% (538/538), done.
Total 4138 (delta 611), reused 854 (delta 482), pack-reused 3058

Checked out Git ref: refs/heads/master
```

## examples

```shell
atmos describe affected
atmos describe affected --verbose=true
atmos describe affected --ref refs/heads/main
atmos describe affected --ref refs/heads/main --format json
atmos describe affected --ref refs/tags/v1.16.0 --file affected.yaml --format yaml
atmos describe affected --ref refs/heads/my-new-branch
atmos describe affected --sha 3a5eafeab90426bd82bf5899896b28cc0bab3073 --file affected.json
atmos describe affected --sha 3a5eafeab90426bd82bf5899896b28cc0bab3073
```

## test

If the current local branch changes a component (`vars` or `env` section) in any YAML stack config files that defines configuration for the component (either inline or via imports), the command shows all the affected components/stacks and the affected section:

```shell
atmos describe affected --ref refs/heads/main
```

```json
[
   {
      "stack": "tenant2-ue2-staging",
      "component_type": "terraform",
      "component": "test/test-component-override-3",
      "affected": "stack.vars"
   },
   {
      "stack": "tenant2-ue2-prod",
      "component_type": "terraform",
      "component": "test/test-component-override-3",
      "affected": "stack.vars"
   },
   {
      "stack": "tenant2-ue2-dev",
      "component_type": "terraform",
      "component": "test/test-component-override-3",
      "affected": "stack.vars"
   },
   {
      "stack": "tenant1-ue2-staging",
      "component_type": "terraform",
      "component": "test/test-component-override-3",
      "affected": "stack.vars"
   },
   {
      "stack": "tenant1-ue2-dev",
      "component_type": "terraform",
      "component": "test/test-component-override-3",
      "affected": "stack.vars"
   },
   {
      "stack": "tenant1-ue2-test-1",
      "component_type": "terraform",
      "component": "test/test-component-override-3",
      "affected": "stack.vars"
   },
   {
      "stack": "tenant1-ue2-prod",
      "component_type": "terraform",
      "component": "test/test-component-override-3",
      "affected": "stack.vars"
   }
]
```

If the current local branch changes any base component (including the `abstract` ones), the command shows all the affected `real` components including all the components derived from the changed base component. For example, if we changed the variables for the `test/test-component` (which is a base for many derived components), the command shows the affected base and all the derived components:

```shell
atmos describe affected --ref refs/heads/main
```

```json
[
  {
      "stack": "tenant1-ue2-dev",
      "component_type": "terraform",
      "component": "test/test-component",
      "affected": "stack.vars"
   },
   {
      "stack": "tenant1-ue2-dev",
      "component_type": "terraform",
      "component": "test/test-component-override",
      "affected": "stack.vars"
   },
   {
      "stack": "tenant1-ue2-dev",
      "component_type": "terraform",
      "component": "test/test-component-override-2",
      "affected": "stack.vars"
   },
   {
      "stack": "tenant1-ue2-dev",
      "component_type": "terraform",
      "component": "test/test-component-override-3",
      "affected": "stack.vars"
   }
]
```

If the current local branch changes the `env` section (ENV vars) for any base or derived components, the command shows the affected `real` components with the `affected_section: "env"`:

```shell
atmos describe affected --ref refs/heads/main
```

```json
[
  {
      "stack": "tenant1-ue2-dev",
      "component_type": "terraform",
      "component": "test/test-component-override-3",
      "affected": "stack.env"
   },
   {
      "stack": "tenant2-ue2-dev",
      "component_type": "terraform",
      "component": "test/test-component-override-3",
      "affected": "stack.env"
   },
   {
      "stack": "tenant2-ue2-staging",
      "component_type": "terraform",
      "component": "test/test-component-override-3",
      "affected": "stack.env"
   },
   {
      "stack": "tenant1-ue2-staging",
      "component_type": "terraform",
      "component": "test/test-component-override-3",
      "affected": "stack.env"
   },
   {
      "stack": "tenant2-ue2-prod",
      "component_type": "terraform",
      "component": "test/test-component-override-3",
      "affected": "stack.env"
   },
   {
      "stack": "tenant1-ue2-prod",
      "component_type": "terraform",
      "component": "test/test-component-override-3",
      "affected": "stack.env"
   },
   {
      "stack": "tenant1-ue2-test-1",
      "component_type": "terraform",
      "component": "test/test-component-override-3",
      "affected": "stack.env"
   }
]
```

If the current local branch changes a Terraform or Helmfile component (in `components/terraform` and `components/helmfile` folders), the command shows all the affected Atmos components/stacks and in the `affected` attribute it shows `"affected": "terraform"` if the Terraform code for the component has been changed, or `"affected": "helmfile" if the Helmfile code for the component has been changed. For example, if we change the `examples/complete/components/terraform/infra/vpc/main.tf` file, it will affect all the Atmos components that provision the  the `vpc` Terraform component:

```shell
atmos describe affected --verbose=true
```

```console
Cloning repo 'https://github.com/cloudposse/atmos' into the temp dir '/var/folders/g5/lbvzy_ld2hx4mgrgyp19bvb00000gn/T/16710736261366892599'

Checking out the HEAD of the default branch

Enumerating objects: 4215, done.
Counting objects: 100% (1157/1157), done.
Compressing objects: 100% (576/576), done.
Total 4215 (delta 658), reused 911 (delta 511), pack-reused 3058

Checked out Git ref: refs/heads/master

Local repo HEAD: 7d37c1e890514479fae404d13841a2754be70cbf refs/heads/add-describe-affected-cli-command
Remote repo HEAD: 40210e8d365d3d88ac13c0778c0867b679bbba69 refs/heads/master

Changed files:
cmd/describe_affected.go
examples/complete/Dockerfile
examples/complete/components/terraform/infra/vpc/main.tf
go.mod
go.sum
internal/exec/describe_affected.go
internal/exec/describe_component.go
internal/exec/describe_stacks.go
internal/exec/utils.go
internal/exec/vendor_utils.go
pkg/config/schema.go
pkg/describe/describe_affected_test.go
website/.nvmrc
website/docs/cli/commands/describe/describe-affected.md

Affected components and stacks:

[
   {
      "stack": "tenant1-ue2-dev",
      "component_type": "terraform",
      "component": "infra/vpc",
      "affected": "component"
   },
   {
      "stack": "tenant1-ue2-dev",
      "component_type": "terraform",
      "component": "vpc",
      "affected": "component"
   },
   {
      "stack": "tenant1-ue2-dev",
      "component_type": "terraform",
      "component": "vpc/new",
      "affected": "component"
   },
   {
      "stack": "tenant1-ue2-test-1",
      "component_type": "terraform",
      "component": "infra/vpc",
      "affected": "component"
   },
   {
      "stack": "tenant1-ue2-test-1",
      "component_type": "terraform",
      "component": "vpc",
      "affected": "component"
   },
   {
      "stack": "tenant2-ue2-prod",
      "component_type": "terraform",
      "component": "infra/vpc",
      "affected": "component"
   },
   {
      "stack": "tenant2-ue2-prod",
      "component_type": "terraform",
      "component": "vpc",
      "affected": "component"
   },
   {
      "stack": "tenant1-ue2-prod",
      "component_type": "terraform",
      "component": "infra/vpc",
      "affected": "component"
   },
   {
      "stack": "tenant1-ue2-prod",
      "component_type": "terraform",
      "component": "vpc",
      "affected": "component"
   },
   {
      "stack": "tenant1-ue2-staging",
      "component_type": "terraform",
      "component": "vpc",
      "affected": "component"
   },
   {
      "stack": "tenant1-ue2-staging",
      "component_type": "terraform",
      "component": "infra/vpc",
      "affected": "component"
   },
   {
      "stack": "tenant2-ue2-dev",
      "component_type": "terraform",
      "component": "infra/vpc",
      "affected": "component"
   },
   {
      "stack": "tenant2-ue2-dev",
      "component_type": "terraform",
      "component": "vpc",
      "affected": "component"
   },
   {
      "stack": "tenant2-ue2-staging",
      "component_type": "terraform",
      "component": "vpc",
      "affected": "component"
   },
   {
      "stack": "tenant2-ue2-staging",
      "component_type": "terraform",
      "component": "infra/vpc",
      "affected": "component"
   }
]
```

<br/>

image

### CODE REVIEW

File: internal/exec/describe_affected.go [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+
+	// Clone the remote repo
+	// https://git-scm.com/book/en/v2/Git-Internals-Git-References
+	// https://github.com/go-git/go-git/tree/master/_examples
+	// https://stackoverflow.com/questions/56810719/how-to-checkout-a-specific-sha-in-a-git-repo-using-golang
+	// https://golang.hotexamples.com/examples/gopkg.in.src-d.go-git.v4.plumbing/-/ReferenceName/golang-referencename-function-examples.html
+
+	u.PrintInfoVerbose(verbose, fmt.Sprintf("\nCloning repo '%s' into a temp dir '%s'", repoUrl, tempDir))
+
+	cloneOptions := git.CloneOptions{
```
Comment (nitrocode) [inline internal/exec/describe_affected.go:142]: Should we also add a way to do a shallow clone for mono repos with many many commits?

File: internal/exec/describe_affected.go [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+	if err != nil {
+		return err
+	}
+
+	sha, err := flags.GetString("sha")
+	if err != nil {
+		return err
+	}
+
+	if ref == "" && sha == "" {
```
Comment (mcalhoun) [inline internal/exec/describe_affected.go:45]: Should we consider setting the default to the head of the default branch if neither is specified rather than throwing an error?

File: internal/exec/describe_affected.go [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+
+	cloneOptions := git.CloneOptions{
+		URL:          repoUrl,
+		NoCheckout:   false,
+		SingleBranch: false,
+	}
+
+	if ref != "" {
+		cloneOptions.ReferenceName = plumbing.ReferenceName(ref)
+		u.PrintInfoVerbose(verbose, fmt.Sprintf("Git ref: %s", ref))
```
Comment (nitrocode) [inline internal/exec/describe_affected.go:150]: This should match line 163 `\nChecking out SHA: %s\n`, no?

```suggestion
		u.PrintInfoVerbose(verbose, fmt.Sprintf("\nChecking out ref: %s\n", ref))
```

File: internal/exec/describe_affected.go (hunk 2 above)
Comment (mcalhoun) [reply internal/exec/describe_affected.go:45]: > Both can be specified

This is also confusing to me...if I specify a SHA shouldn't I get all changes on my current branch since that SHA? What happens if I specify both REF and SHA?

File: internal/exec/describe_affected.go [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+
+	affected := findAffected(currentStacks, remoteStacks)
+	return affected, nil
+}
+
+func findAffected(currentStacks map[string]any, remoteStacks map[string]any) []cfg.Affected {
+	res := []cfg.Affected{}
+
+	for stackName, stackSection := range currentStacks {
+		if stackSectionMap, ok := stackSection.(map[string]any); ok {
```
Comment (nitrocode) [inline internal/exec/describe_affected.go:213]: Is there a way we can simplify these statements? This seems like quite a nest 😬

File: internal/exec/describe_affected.go [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+
+	if ref == "" && sha == "" {
+		return fmt.Errorf("invalid flag. Either '--ref' or '--sha' is required. Both can be specified")
+	}
+
+	format, err := flags.GetString("format")
+	if err != nil {
+		return err
+	}
+	if format != "" && format != "yaml" && format != "json" {
```
Comment (mcalhoun) [inline internal/exec/describe_affected.go:53]: nit: add a blank line before

File: internal/exec/describe_affected.go [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+		return fmt.Errorf("invalid '--format' flag '%s'. Valid values are 'yaml' (default) and 'json'", format)
+	}
+	if format == "" {
```
Comment (mcalhoun) [inline internal/exec/describe_affected.go:56]: nit: add a blank line before

File: internal/exec/describe_affected.go [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+						if componentSection, ok := compSection.(map[string]any); ok {
+							// Skip abstract components
+							if metadataSection, ok := componentSection["metadata"].(map[any]any); ok {
+								if metadataType, ok := metadataSection["type"].(string); ok {
+									if metadataType == "abstract" {
+										continue
+									}
+								}
+							}
+							// Check `vars` section
```
Comment (nitrocode) [inline internal/exec/describe_affected.go:226]: How come the other `metadata` keys and `settings` keys are not checked? Wouldn't changes to those parent keys also affect the stacks?

File: internal/exec/describe_affected.go [hunk 8]
```diff
@@ -0,0 +1,326 @@
...
+			if err != nil {
+				return err
+			}
+		} else {
+			err = u.WriteToFileAsJSON(file, affected, 0644)
+			if err != nil {
+				return err
+			}
+		}
+	}
```
Comment (mcalhoun) [inline internal/exec/describe_affected.go:99]: consider moving this to a case statement with a default that returns an error with unsupported format

File: internal/exec/describe_stacks.go [hunk 9]
```diff
@@ -73,11 +73,54 @@ func ExecuteDescribeStacksCmd(cmd *cobra.Command, args []string) error {
 		sections = strings.Split(sectionsCsv, ",")
 	}
 
-	stacksMap, err := FindStacksMap(cliConfig)
+	finalStacksMap, err := ExecuteDescribeStacks(cliConfig, filterByStack, components, componentTypes, sections)
 	if err != nil {
 		return err
 	}
 
+	if format == "yaml" {
```
Comment (nitrocode) [inline internal/exec/describe_stacks.go:81]: This same code is in `ExecuteDescribeAffectedCmd`. Can we create a new function in a utils.go or equivalent and use the function in both places?

**Review by nitrocode — DISMISSED** [review]
Looks good to me. I had a couple nitpicks. I had one comment regarding the settings and metadata keys. Those might be nice-to-have but could be added in a subsequent PR too.

Nice job!

File: internal/exec/describe_affected.go [hunk 10]
```diff
@@ -0,0 +1,326 @@
...
+
+	// Get the origin URL of the current repo
+	repoUrl, err := gitconfig.OriginURL()
+	if err != nil {
+		return nil, err
+	}
+	if repoUrl == "" {
+		return nil, errors.New("the current repo is not a Git repository. Check that it was initialized and has '.git' folder")
+	}
+
```
Comment (mcalhoun) [inline internal/exec/describe_affected.go:120]: Is it possible that a valid git repo doesn't have an origin set?

File: internal/exec/describe_affected.go [hunk 11]
```diff
@@ -0,0 +1,326 @@
...
+	// Create a temp dir to clone the remote repo to
+	tempDir, err := os.MkdirTemp("", strconv.FormatInt(time.Now().Unix(), 10))
```
Comment (mcalhoun) [inline internal/exec/describe_affected.go:122]: nit: consider adding a prefix or suffix just in case someone else uses this technique

File: internal/exec/describe_affected.go [hunk 12]
```diff
@@ -0,0 +1,326 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	defer func(path string) {
+		err := os.RemoveAll(path)
+		if err != nil {
+			u.PrintError(err)
+		}
+	}(tempDir)
```
Comment (mcalhoun) [inline internal/exec/describe_affected.go:132]: nit: consider moving this to a separate function like `removeTempDir`. It makes the main body more readable with a clear name (without having to read that code).

File: internal/exec/describe_affected.go [hunk 13]
```diff
@@ -0,0 +1,326 @@
...
+		URL:          repoUrl,
+		NoCheckout:   false,
+		SingleBranch: false,
+	}
+
+	if ref != "" {
+		cloneOptions.ReferenceName = plumbing.ReferenceName(ref)
+		u.PrintInfoVerbose(verbose, fmt.Sprintf("Git ref: %s", ref))
+	}
+	if verbose {
```
Comment (mcalhoun) [inline internal/exec/describe_affected.go:152]: nit: add a blank line

File: internal/exec/describe_affected.go [hunk 14]
```diff
@@ -0,0 +1,326 @@
...
+		cloneOptions.Progress = os.Stdout
+	}
+
+	repo, err := git.PlainClone(tempDir, false, &cloneOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check if a commit SHA was provided and checkout the repo at that commit SHA
```
Comment (mcalhoun) [inline internal/exec/describe_affected.go:161]: If you specified both a `ref` and a `sha` as the docs indicate is possible, won't this result in doing two different clones? Either (a) you shouldn't be able to specify both or (b) if sha takes precedence, then this should probably be an else if

**Review by mcalhoun — CHANGES_REQUESTED** [review]
Overall it looks like a good first pass. I left a few comments and a few nits.

The one thing that I think is missing, unless I missed it, is the detection of any of the terraform (or helmfile code) for a component changing. 

For example if we had a component vpc and we changed an output but didn't change the stack config, how would that be detected?

**Comment by aknysh** [comment]
> Overall it looks like a good first pass. I left a few comments and a few nits.
> 
> The one things that I think is missing, unless I missed it, is the detection of any of the terraform (or helmfile code) for a component changing.
> 
> For example if we had a component vpc and we changed an output but didn't change the stack config, how would that be detected?

This is not supported currently.
And it prob can be done by using plain `git` commands (they are simple) instead of embedding this into atmos.
The plain git commands are not possible to use for atmos stacks since there is a lot of processing involved.

Let me think/check if that can be done with `go-git`

**Comment by osterman** [comment]
> > For example if we had a component vpc and we changed an output but didn't change the stack config, how would that be detected?
> This is not supported currently.

@aknysh we need to include this. It's too complicated to figure it out outside of this and still need to map it back to the affected stacks. It can be a new PR, but should be implemented natively in this command.

**Comment by aknysh** [comment]
> Looks good to me. I had a couple nitpicks. I had one comment regarding the settings and metadata keys. Those might be nice-to-have but could be added in a subsequent PR too.
> 
> Nice job!

I've included `settings` and `metadata` into the comparison as well (not that we need it, but just in case).

**Comment by mcalhoun** [comment]
> This is not supported currently.
And it prob can be done by using plain git commands (they are simple) instead of embedding this into atmos.
The plain git commands are not possible to use for atmos stacks since there is a lot of processing involved.

It definitely has to be done in atmos, because we have to tie it back to each affected stack, including all the inherited components, etc. So if the `vpc` component's terraform is modified, then we need all the components and inherited components that are affected by that change.

File: internal/exec/describe_affected.go (hunk 2 above)
Comment (mcalhoun) [reply internal/exec/describe_affected.go]: This is still somewhat confusing to me...with regular git, if you checkout a commit, the branch is not necessary. It checks out the repository's state at that specific sha. The sha IS a ref just like the HEAD is an alias for whatever sha the main branch is currently pointing to. It seems to me that `ref` and `sha` shouldn't ever both be specified.

File: internal/exec/describe_affected.go (hunk 11 above)
Comment (mcalhoun) [reply internal/exec/describe_affected.go:100]: I was suggesting that the current epoch time may not be unique enough. It was a nit anyway, and easy enough to fix if we run into an issue.

**Comment by aknysh** [comment]
> > This is not supported currently.
> > And it prob can be done by using plain git commands (they are simple) instead of embedding this into atmos.
> > The plain git commands are not possible to use for atmos stacks since there is a lot of processing involved.
> 
> It definitely has to be done in atmos, because we have to tie it back to each affected stack, including all the inherited components, etc. So if the `vpc` component's terraform is modified, then we need all the components and inherited components that are affected by that change.

it's implemented, see the PR description

File: website/docs/cli/commands/describe/describe-affected.md [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+- `component` is the affected Atmos component in the stack
+- `component_type` is the type of the affected Atmos component (`terraform` or `helmfile`)
+- `affected` shows what was changed for the component. The possible values are:
+
+  - `vars` - the `vars` component section in the stack config has been modified
+  - `env` - the `env` component section in the stack config has been modified
+  - `settings` - the `settings` component section in the stack config has been modified
+  - `metadata` - the `metadata` component section in the stack config has been modified
+  - `terraform` - the Terraform component (Terraform files) that the affected Atmos component provisions has been changed
+  - `helmfile` - the Helmfile component (Helmfile files) that the affected Atmos component provisions has been changed
```
Comment (osterman) [inline website/docs/cli/commands/describe/describe-affected.md:85]: IMO, this should just be consolidated to `component`, as `component_type` will be the indicator of if it's `terraform` or `helmfile`

Comment (osterman) [reply website/docs/cli/commands/describe/describe-affected.md:85]: Consider changing affected to be disambiguated a little bit: `stack.vars` or `stack.env`.

File: website/docs/cli/commands/describe/describe-affected.md [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+```shell
+atmos describe affected
+atmos describe affected --verbose=true
+atmos describe affected --ref refs/heads/main
+atmos describe affected --ref refs/heads/my-new-branch --verbose=true
+atmos describe affected --ref refs/heads/main --format json
+atmos describe affected --ref refs/tags/v1.16.0 --file affected.yaml --format yaml
+atmos describe affected --sha 3a5eafeab90426bd82bf5899896b28cc0bab3073 --file affected.json
+atmos describe affected --sha 3a5eafeab90426bd82bf5899896b28cc0bab3073
+```
```
Comment (osterman) [inline website/docs/cli/commands/describe/describe-affected.md:46]: Ideally, we should break each of these apart and explain what it does. 

for example, 

~~~
This command will describe all affected stacks relative to the default branch
```
atmos describe affected
```

This command will describe all affected stacks relative to the main branch provide verbose output including every step performed to arrive at the conclusion. 

```
atmos describe affected --verbose=true
```

~~~

File: website/docs/cli/commands/describe/describe-affected.md [hunk 17]
```diff
@@ -0,0 +1,112 @@
...
+:::
+
+<br/>
+
+:::info
+For the first commit, the command assumes that the repo root is a Git checkout.
+
+The second commit is specified on the command line by using the `--ref` or `--sha` flags.
+
+If the flags are not provided, it will clone the HEAD of the default branch.
```
Comment (osterman) [inline website/docs/cli/commands/describe/describe-affected.md:20]:
```suggestion
If the flags are not provided, the `ref` be the default branch (e.g. `main`) and the `sha` will point to the `HEAD` of the branch.
```

Comment (osterman) [reply website/docs/cli/commands/describe/describe-affected.md:20]: The fact that we clone it, will always be the case. I think we should describe our logic of how this works in plain English.

This command works by:
1. Cloning the branch (`--ref`) and checking out the commit (`--sha`) of the target branch. 
2. Deep merge all configurations for both the current working branch and the target branch
3. Compare each section of the stack configuration looking for differences
4. Look for changes in the component directories
5. De-dupe affected stacks
6. Output a JSON document consisting of a list of affected stacks and what caused it to be affected

**Review by osterman — APPROVED** [review]

File: website/docs/cli/commands/describe/describe-affected.md [hunk 18]
```diff
@@ -0,0 +1,174 @@
...
+to the `HEAD` of the branch.
+
+The command works by:
+
+- Cloning the branch (`--ref`) or checking out the commit (`--sha`) of the remote target branch
+- Deep merging all stack configurations for both the current working branch and the target branch
+- Looking for changes in the component directories
+- Comparing each section of the stack configuration looking for differences
+- Output a JSON or YAML document consisting of a list of affected components and stacks and what caused it to be affected
+
```
Comment (osterman) [inline website/docs/cli/commands/describe/describe-affected.md:36]: @aknysh if both the component and stacks are affected, which will be output as the `affected` entity?

File: website/docs/cli/commands/describe/describe-affected.md [hunk 19]
```diff
@@ -0,0 +1,174 @@
...
+<br/>
```
Comment (osterman) [inline website/docs/cli/commands/describe/describe-affected.md:37]:
```suggestion
Since Atmos first checks the component folders for changes, if it finds any affected files, it will mark all related stacks as affected. Atmos will then skip evaluating those stacks for differences since we already know that they are affected.

<br/>
```

**Review by mcalhoun — DISMISSED** [review]
LGTM!

**Review by Benbentwo — APPROVED** [review]
LGTM!

**Review by osterman — APPROVED** [review]
